superset-frontend/src/views/dashboardList/DashboardList.tsx

```diff
@@ -508,71 +510,68 @@ class DashboardList extends React.PureComponent<Props, State> {
       dashboardToEdit,
     } = this.state;
     return (
-      <div className="container welcome">
-        <Panel>
-          <Panel.Body>
-            <ConfirmStatusChange
-              title={t('Please confirm')}
-              description={t(
-                'Are you sure you want to delete the selected dashboards?',
-              )}
-              onConfirm={this.handleBulkDashboardDelete}
-            >
-              {confirmDelete => {
-                const bulkActions = [];
-                if (this.canDelete) {
-                  bulkActions.push({
-                    key: 'delete',
-                    name: (
-                      <>
-                        <i className="fa fa-trash" /> Delete
-                      </>
-                    ),
-                    onSelect: confirmDelete,
-                  });
-                }
-                if (this.canExport) {
-                  bulkActions.push({
-                    key: 'export',
-                    name: (
-                      <>
-                        <i className="fa fa-database" /> Export
-                      </>
-                    ),
-                    onSelect: this.handleBulkDashboardExport,
-                  });
-                }
-                return (
+      <>
```

Comment:
it seems like a lot of code is the same as in the above file, is there any way we can pull it out into its own component and reuse it?

Reply:
this is where it gets tricky, the code is largely the same but there are a few differences that make it difficult to isolate to a single component. It also couples the code so that changes in one view could result in unexpected changes in another view. I am planning to revisit this question once some of the design on these listviews has settled and is not likely to change drastically.

Reply:
Something that comes to mind as a flexible solution that would allow customization across the different views is a BaseListViewPage component that these could inherit from and override certain methods as needed. This would remove some of the code redundancies while allowing flexibility in overriding certain behavior. I generally try to avoid inheritance in JS when possible, but it seems like there's a case for it here. Thoughts? 

Another approach might be to move this into hooks, ie `usePermissions`, `useFetchListData`, etc. though I'm not convinced this will really remove that much code redundancy